BringgSDK.spec.js

```diff
@@ -39,6 +39,57 @@ describe('BringgSDK', function () {
     expect(BringgSDK._setCredentials).toBeDefined();
   });
 
+  describe('initializeBringg', function() {
+    beforeEach(function() {
+      // Don't call to google on tests
+      spyOn(BringgSDK, '_setDriverActivity').and.stub();
+
+      // Don't connect to realtime
+      spyOn(window, 'io').and.returnValue({ on: function(){} });
+
+      // Mock getJSON for getSharedConfig
+      spyOn($, 'getJSON').and.callFake(function(url, callback) {
+        callback({});
+        return { error: function(){} };
+      });
+    });
+
+    it('connects to the right url by developer token', function(done) {
+      var config = {
+        token: "ew1_MySecretToken",
+        share_uuid: "shared"
+      };
+
+      function onInit() {
+        expect(window.io).toHaveBeenCalled();
+
+        // make sure we are connected to europe since
+        // token starts with "ew1_"
+        var socketUrl = window.io.calls.first().args[0];
+        expect(socketUrl).toEqual("https://eu1-realtime.bringg.com");
+
+        done();
+      }
+
+      BringgSDK.initializeBringg(config, onInit, function onFailure() {
+        done(new Error('Failed!'));
+      });
+    });
+
+    describe('_setCredentials', function() {
+      it('figures out the developer access token', function() {
```

Comment:
please spy and verify that setupConfigByToken is being called for each (or otherwise test the various cases there (i.e when you return for invalid input for example).

also please add a test case for else (that this is not being called.

lastly, should be nicer to separate each test with different token to a different test case (in case 1 fails it easier to find and debug).